### PR TITLE
Add support for app.sh script so have easy control over how application launched.

### PR DESCRIPTION
**Pull request by GrahamDumpleton**
This address issue https://github.com/sclorg/s2i-python-container/issues/126.

Note that you should merge change https://github.com/sclorg/s2i-python-container/pull/165 first as this depends on the test case added in that for uWSGI, which uses an ``app.sh`` file executed from an ``app.py`` file. When the change here is then merged, the ``app.sh`` file in the test will take precedence and the ``app.sh`` will be executed directly with ``app.py`` being bypassed.

Note that this change doesn't do anything about ``APP_HOME`` as existing code didn't do anything with ``APP_HOME`` for ``app.py`` either. Arguably it should, but this is being dealt with in separate issues and PR.

Also note that although the change uses ``APP_SCRIPT`` and refers to it as application script file, encouraging it as being used for a shell script, technically it could be any executable script file, or even a compiled application. A generic name for the environment variable could be used such as ``APP_PROGRAM`` if really want to reflect that, but probably better to encourage it being just used for a shell script which then in turn can execute anything.

### CODE REVIEW

**Comment by pkubatrh** [comment]
#165 has been merged, please rebase so that we can use the tests in CI

**Comment by pkubatrh** [comment]
[test]

File: 2.7/README.md [hunk 1]
```diff
@@ -197,10 +203,19 @@ following ways, in precedence order:
...
+  Python script via the `APP_FILE` environment variable, defaulting to a
+  file named `app.py` if it exists. The script is passed to a regular
+  Python interpreter to launch your application.
+
+* **Application script file**
+
+  This is the most general way of executing your application. It will be
+  used in the case where you specify a path to an executable script file
+  via the `APP_SCRIPT` environment variable, defaulting to a file named
+  `app.sh` if it exists. The script is executed directory to launch your
```
Comment (pkubatrh) [inline 2.7/README.md]: I guess directory -> directly?

**Comment by pkubatrh** [comment]
One more thing - this part should also get changed to accommodate for the new APP_SCRIPT variable
https://github.com/GrahamDumpleton/s2i-python-container/blob/7d09b2ba22c60a1c012fed00b5f11cc13375e7f6/2.7/s2i/bin/run#L102

**Comment by GrahamDumpleton** [comment]
Added ``APP_SCRIPT`` environment variable to final error message.

**Comment by hhorak** [comment]
LGTM.

**Comment by hhorak** [comment]
[test]

**Pull request merged**